odoo/addons/base/models/ir_actions.py

```diff
@@ -229,9 +229,22 @@ def _get_action_dict(self):
         self.ensure_one()
         readable_fields = self._get_readable_fields()
         if (self.sudo().type == "ir.actions.act_window"):
-            embedded_actions_readable_fields = self.env["ir.actions.embedded"]._get_readable_fields()
-            dict_readable_fields = {key: {} for key in readable_fields}
-            dict_embedded_actions_readable_fields = {key: {} for key in embedded_actions_readable_fields}
+            IrEmbededActions = self.env["ir.actions.embedded"]
+            embedded_actions_readable_fields = IrEmbededActions._get_readable_fields()
+
+            def get_specification(field):
+                specification = {}
+                if field.relational:
+                    specification["fields"] = {}
+                    if field.type == "many2one":
+                        specification["fields"]['display_name'] = {}
+                return specification
+
+            dict_readable_fields = {key: get_specification(self._fields[key]) for key in readable_fields}
+            dict_embedded_actions_readable_fields = {
+                key: get_specification(IrEmbededActions._fields[key])
+                for key in embedded_actions_readable_fields
+            }
             return self.sudo().web_read({
```

Comment:
Coucou @luve-odoo @ryv-odoo and @aab-odoo 

I added a fixup commit because the object returned by `read` and `web_read` is slightly different.
the `read` will return a tuple for many2one compared to an int for `web_read` if we don't fetch the `display_name`.

My first assumption was to ask to `web_read` to fetch the `display_name` to get the same result. BUT, `web_read` will return a dict. 😅

So, either we add this and we change the behavior in action_service or we could just change the action_service.js to do what we did in the last commit in that PR. 🙂

Reply:
@xavierbol That's fine. Could you add a // TODO comment s.t. we remove the support of search_view_id given as an Array?